### DISCUSSION TITLE
ApolloPersistedQuerySupport - GraphQL Exception Query parameter not found in GET request

### DISCUSSION THREAD

**Discussion by codingpierogi** [graphql-java/graphql-java#2647] [Q&A]
Hi,

I'm trying to implement ApolloPersistedQuerySupport in a Spring Boot application and I'm stuck on the GraphQL Exception "Query parameter not found in GET request.".

My React application is using Apollo Client and issues the following HTTP request:

`GET http://localhost:8080/graphql?variables=%7B%22locale%22%3A%22en%22%7D&extensions=%7B%22persistedQuery%22%3A%7B%22version%22%3A1%2C%22sha256Hash%22%3A%229bfd7a6b058c1c1a2b744eedeb6eff2bdcf231edfe67ac172c4177eb26a238c1%22%7D%7D`

The GraphQL Server is then throwing a GraphQL Exception from GraphQLGetInvocationInputParser. Relevant code below:

` 
String query = request.getParameter("query");
if (query == null) {
  throw new GraphQLException("Query parameter not found in GET request");
}
`

Is there anymore documentation around this feature Execution > Query Caching? Any open source projects that implement ApolloPersistedQuerySupport?

Any help greatly appreciated.

Thanks!

**Comment by jord1e** [comment]
What framework are you using to handle HTTP

**Reply by codingpierogi** [reply]
Relevant dependencies are:

npm: @apollo/client@^3.5.5
maven: org.springframework.boot:spring-boot:2.5.7
maven: com.graphql-java-kickstart:graphql-spring-boot-starter:12.0.0

**Reply by jord1e** [reply]
The correct place to ask for help would be https://github.com/graphql-java-kickstart/graphql-spring-boot

I think the problem lies in the fact that you have not registered the ApolloPersistedQuerySupport PreparsedDocumentProvider

https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/execution/preparsed/persisted/ApolloPersistedQuerySupport.java

You could probably autowire the bean
https://github.com/graphql-java-kickstart/graphql-spring-boot/blob/13cd484f913fe7885ff5d5bc5883b1e9544d4ca6/graphql-spring-boot-autoconfigure/src/main/java/graphql/kickstart/spring/web/boot/GraphQLWebAutoConfiguration.java#L125

```java
@Configuration
public class ApqConfig {
  @Bean
  public PreparsedDocumentProvider apqProvider() {
    return new ApolloPersistedQuerySupport(InMemoryPersistedQueryCache().newInMemoryPersistedQueryCache().build());
  }
}
```

I have not tried this and can't help you any further. The kickstart issues are probably the best place to ask

Please [mark this as an answer](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer) if it helped you.

**Reply by codingpierogi** [reply]
Unfortunately that suggestion didn't work...got the same exception. Thanks for taking the time to respond! I will reach out to graphql-java-kickstart. Thanks
